process/smartContract/hooks/blockChainHook.go

```diff
@@ -762,6 +827,28 @@ func (bh *BlockChainHookImpl) EpochConfirmed(epoch uint32, _ uint64) {
 	}
 }
 
+// GasScheduleChange sets the new gas schedule where it is needed
+func (bh *BlockChainHookImpl) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
```

Comment:
refactoring suggestion to avoid nested mutex lockings that is a very dangerous business:
```GO
// GasScheduleChange sets the new gas schedule where it is needed
func (bh *BlockChainHookImpl) GasScheduleChange(gasSchedule map[string]map[string]uint64) {
	maxPerTransaction := bh.getMaxPerTransactionValues(gasSchedule)
	if maxPerTransaction == nil{
		log.Error("maxPerTransaction definition is missing in the current gas schedule, using old values")
		return
	}

	bh.mutCounters.Lock()
	defer mutCounters.Unlock()
	
	bh.maxBuiltInCallsPerTx = maxPerTransaction["MaxBuiltInCallsPerTx"]
	bh.maxNumberOfTransfersPerTx = maxPerTransaction["MaxNumberOfTransfersPerTx"]
	bh.maxNumberOfTrieReadsPerTx = maxPerTransaction["MaxNumberOfTrieReadsPerTx"]
}

func (bh *BlockChainHookImpl) getMaxPerTransactionValues(gasSchedule map[string]map[string]uint64) map[string]uint64{
	bh.mutGasLock.Lock()
	defer bh.mutGasLock.Unlock()

	maxPerTransaction := gasSchedule[common.MaxPerTransaction]
	if maxPerTransaction == nil {
		return nil
	}
	
	result := make(map[string]uint64)
	for key, value := range maxPerTransaction{
		result[key] = value
	}
	
	return result
}
```

Reply:
done